CoreFoundation/Base.subproj/CFPlatform.c

```diff
@@ -1495,6 +1499,15 @@ CF_CROSS_PLATFORM_EXPORT int _CFThreadSetName(_CFThreadRef thread, const char *_
 CF_CROSS_PLATFORM_EXPORT int _CFThreadGetName(char *buf, int length) {
 #if TARGET_OS_MAC
     return pthread_getname_np(pthread_self(), buf, length);
+#elif TARGET_OS_ANDROID
+    // Android did not get pthread_getname_np until API 26, but prctl seems to
+    // return at least 15 chars of the name.
+    char *buffer[16] = {0};
+    if (prctl(PR_GET_NAME, buffer, 0, 0, 0) != 0) {
+        return -1;
+    }
+    memcpy(buf, buffer, MIN(strnlen(buffer, 16), length));
```

Comment:
Dont you need to copy the `NUL` terminator as well. Should it not be:
```
size_t sz = MIN(strnlen(buffer, 16), length - 1);
memcpy(buf, buffer, sz);
buf[sz] = '\0';
```

Reply:
👍